plotly/tools.py

```diff
@@ -3333,30 +3333,39 @@ def _trisurf(x, y, z, simplices, show_colorbar, colormap=None,
                                                       max_mean_dists)
                 facecolor.append(color)
 
-        # Make sure we have arrays to speed up plotting
+        # Make sure facecolor is a list so output is consistent across Pythons
         facecolor = list(facecolor)
         ii, jj, kk = simplices.T
 
-        # make a colorscale from the colors
-        colorscale = FigureFactory._make_colorscale(colormap)
-        colorscale = FigureFactory._convert_colorscale_to_rgb(colorscale)
-
         triangles = graph_objs.Mesh3d(x=x, y=y, z=z, facecolor=facecolor,
                                       i=ii, j=jj, k=kk, name='')
 
-        colorbar = graph_objs.Mesh3d(x=[0, 1], y=[0, 1], z=[0, 1],
-                                     colorscale=colorscale,
-                                     intensity=[0, 1],
-                                     showscale=True)
+        if not isinstance(mean_dists[0], str) and show_colorbar is True:
+            # make a colorscale from the colors
+            colorscale = FigureFactory._make_colorscale(colormap)
+            colorscale = FigureFactory._convert_colorscale_to_rgb(colorscale)
+
+            colorbar = graph_objs.Scatter3d(
+                x=x[:2],
+                y=y[:2],
+                z=z[:2],
+                mode='markers',
+                marker=dict(
+                    size=0.1,
+                    color=[min_mean_dists, max_mean_dists],
+                    colorscale=colorscale,
+                    showscale=True),
+                hoverinfo='None',
+                showlegend=False
+            )
 
         # the triangle sides are not plotted
-        if plot_edges is not True:
-            if show_colorbar is True:
+        if plot_edges is False:
+            if not isinstance(mean_dists[0], str) and show_colorbar is True:
```

Comment:
🐄 you use this `not isinstance(mean_dists[0], str)` twice, can you assign it to a meaningful variable name to make the code easier to understand?